torch/csrc/distributed/rpc/future_message.h

```diff
@@ -23,6 +23,15 @@ struct TORCH_API FutureMessage final {
   // If completed() the callback will be invoked in-place.
   void addCallback(const Callback& callback);
 
+  // Get the destination rank of this future.
+  inline int dst() const {
```

Comment:
cc @zhaojuanmao @mrshenli I had to add the dest rank to the future object so that `hasPendingMessage()` in pg agent can return correctly (it checks to ensure that all send/recv counts are consistent). I'm not sure if we want to add this to the future, is there a better way to do this?

Reply:
This is now removed.